[Docs] Add details for wait_for_async_insert (#63005)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

### wait_for_async_insert {#wait-for-async-insert}

> Enables or disables waiting for processing of asynchronous insertion. If enabled, server will return `OK` only after the data is inserted. Otherwise, it will return `OK` even if the data has not yet been inserted.

Reviewer: ```suggestion
Enables or disables waiting for processing of asynchronous insertion. If enabled, server will return `OK` only after the data is inserted. Otherwise, it will return `OK` as soon it has received the data, but it might still fail to parse or insert it later (You can check in system.asynchronous_insert_log)
```

> If you are using asynchronous inserts, we also recommend enabling [`async_insert`](#async-insert).

Reviewer: ```suggestion
If you want to use asynchronous inserts, we need to also enable [`async_insert`](#async-insert).
```
